lib/tapioca/generators/gem.rb

```diff
@@ -0,0 +1,372 @@
+              end
+
+              gem = T.must(bundle.gem(gem_name))
+              compile_gem_rbi(gem)
+              add(filename)
+
+              puts
+            end
+          end
+
+          anything_done = true
+        end
+
+        puts
+
+        anything_done
+      end
+
+      sig { returns(T::Array[String]) }
+      def removed_rbis
+        (existing_rbis.keys - expected_rbis.keys).sort
+      end
+
+      sig { params(gem_name: String).returns(Pathname) }
+      def existing_rbi(gem_name)
+        gem_rbi_filename(gem_name, T.must(existing_rbis[gem_name]))
+      end
+
+      sig { returns(T::Array[String]) }
+      def added_rbis
+        expected_rbis.select do |name, value|
+          existing_rbis[name] != value
+        end.keys.sort
+      end
+
+      sig { params(gem_name: String).returns(Pathname) }
+      def expected_rbi(gem_name)
+        gem_rbi_filename(gem_name, T.must(expected_rbis[gem_name]))
+      end
+
+      sig { params(gem_name: String).returns(T::Boolean) }
+      def gem_rbi_exists?(gem_name)
+        existing_rbis.key?(gem_name)
+      end
+
+      sig { params(cause: Symbol, files: T::Array[String]).returns(String) }
+      def build_error_for_files(cause, files)
+        filenames = files.map do |file|
+          config.outpath / file
+        end.join("\n  - ")
+
+        "  File(s) #{cause}:\n  - #{filenames}"
+      end
+
+      sig { params(old_filename: Pathname, new_filename: Pathname).void }
+      def move(old_filename, new_filename)
+        say("-> Moving: #{old_filename} to #{new_filename}")
+        old_filename.rename(new_filename.to_s)
+      end
+
+      sig { params(filename: Pathname).void }
+      def add(filename)
+        say("++ Adding: #{filename}")
+      end
+
+      sig { params(gem_name: String, version: String).returns(Pathname) }
+      def gem_rbi_filename(gem_name, version)
+        config.outpath / "#{gem_name}@#{version}.rbi"
+      end
+
+      sig { returns(T::Hash[String, String]) }
+      def existing_rbis
+        @existing_rbis ||= Pathname.glob((config.outpath / "*@*.rbi").to_s)
+          .map { |f| T.cast(f.basename(".*").to_s.split("@", 2), [String, String]) }
+          .to_h
+      end
+
+      sig { returns(T::Hash[String, String]) }
+      def expected_rbis
+        @expected_rbis ||= bundle.dependencies
+          .reject { |gem| config.exclude.include?(gem.name) }
+          .map { |gem| [gem.name, gem.version.to_s] }
+          .to_h
+      end
+
+      sig { params(message: String, color: T.any(Symbol, T::Array[Symbol])).void }
+      def say_error(message = "", *color)
+        force_new_line = (message.to_s !~ /( |\t)\Z/)
+        # NOTE: This is a hack. We're no longer subclassing from Thor::Shell::Color
+        # so we no longer have access to the prepare_message call.
+        # We should update this to remove this.
+        buffer = shell.send(:prepare_message, *T.unsafe([message, *T.unsafe(color)]))
```

Comment:
Similarly, since we no longer subclass `Thor::Shell::Color`, the `prepare_message` protected method is no longer available to us. I had to do a small hack here to make it work as it did previously, but will need to re-evaluate in a future PR. 

My hope is to extract this kind of logic out to a `Logger` class that can abstract this away, and handle it for us.